reporter/src/main/kotlin/reporters/StaticHtmlReporter.kt

```diff
@@ -629,14 +631,17 @@ class StaticHtmlReporter : Reporter {
 }
 
 private fun EM.provenanceLink(provenance: Provenance?) {
-    provenance?.sourceArtifact?.let { artifact ->
-        +" (from "
-        a(href = artifact.url) { +"artifact" }
-        +")"
-    } ?: provenance?.vcsInfo?.let { vcs ->
-        +" (from "
-        a(href = vcs.url) { +"VCS" }
-        +")"
+    when (provenance) {
```

Comment:
Interesting, why doesn't this `when` need to be exhaustive? Because it's nullable? How about making the function take a `KnownProvenance` an in the caller do

    provenanceLink(firstLicenseLocation?.provenance as? KnownProvenance)

Also this could become a function expression.

Reply:
It's not exhaustive because it does not return anything. It only acts in some cases and ignores the others.

Reply:
However, I still get this IDE hint:

> 'when' expression on sealed classes is recommended to be exhaustive, add 'UnknownProvenance' branch or 'else' branch instead

Reply:
I chose to ignore that recommendation because it does not make sense to me in this case. And I didn't want to make the code more complex just to get rid of the warning. If you really want to get rid of the warning I'd be fine with adding a `else -> {}` branch.

Reply:
I was about to say we could also suppress the hint as explained [here](https://stackoverflow.com/a/61756922/1127485), but for some reason it does not work for me:

Reply:
I decided to use `if`/`else if` instead because it was shorter and has less nesting.